### PR TITLE
Added login controller and session handling

### PR DESCRIPTION
**Pull request by AnnaHsu1**
Implemented the backend controller for the login feature of issue #36

### CODE REVIEW

**Review by imran1289-ah — APPROVED** [review]
Looks good in general, although I am not sure what the "getUserByEmail" does. Can someone just briefly clarify what it does.  Also a suggestion from me is maybe start writing some comments on the code as it will help us for future sprints when someone else will be working on the backend

File: src/server/controllers/usersController.js [hunk 1]
```diff
@@ -1,12 +1,33 @@
 
 const User = require('../models/user');
+var sesh;
```
Comment (Aboudik1452) [inline src/server/controllers/usersController.js:3]: I'm not sure what's the purpose of the `sesh` variable which isn't being used.

File: src/server/models/user.js [hunk 2]
```diff
@@ -21,8 +21,17 @@ const userSchema = new Schema({
         type: String,
         required: true
     }
-
-
 });
 
-module.exports = mongoose.model("PublicUser", userSchema);
\ No newline at end of file
+// user.js
```
Comment (Aboudik1452) [inline src/server/models/user.js:26]: Once the front-end sign up page is created, we can remove the commented lines.

File: src/server/controllers/usersController.js [hunk 3]
```diff
@@ -1,12 +1,33 @@
...
+            res.status(200).json(user)
+        }
+        
+        else {
+            return res.status(404).json({
+                errors: [{ user: "not found" }],
+              });
+            }   
+    })
 }
```
Comment (Aboudik1452) [inline src/server/controllers/usersController.js:31]: @imran1289-ah , I think that `verifyUser` is what `getUserByEmail` should be. The function code reflects that.
On this file, the `getUserByEmail` function is commented out from the exports.

Feel free to correct me if I'm wrong @AnnaHsu1 or @Resident-png 🙂

**Review by Aboudik1452 — CHANGES_REQUESTED** [review]

File: src/server/controllers/usersController.js (hunk 1 above)
Comment (AnnaHsu1) [reply src/server/controllers/usersController.js:3]: This is for the session management after logging in, but it is not implemented yet

File: src/server/models/user.js (hunk 2 above)
Comment (AnnaHsu1) [reply src/server/models/user.js:26]: Will do

File: src/server/controllers/usersController.js (hunk 3 above)
Comment (AnnaHsu1) [reply src/server/controllers/usersController.js:31]: We decided to separated verifyUser, which is for the login, and getUserbyEmail, which is a  more general route for fetching users. It is not currently used but we decided to leave it here for the future.

**Review by Aboudik1452 — APPROVED** [review]

**Pull request merged**